tests/cucumber/steps/stepdefs.js

```diff
@@ -2322,3 +2344,239 @@ When('I add a rekeyTo field with the private key algorand address', function ()
 When('I set the from address to {string}', function (from) {
     this.txn["from"] = from;
 });
+
+////////////////////////////////////
+// begin application test helpers
+////////////////////////////////////
+
+Given('a signing account with address {string} and mnemonic {string}', function (address, mnemonic) {
+    this.signingMnemonic = mnemonic
+});
+
+function operationStringToEnum(inString) {
+    switch(inString) {
+        case "call":
+            return algosdk.OnApplicationComplete.NoOpOC;
+        case "create":
+            return algosdk.OnApplicationComplete.NoOpOC;
+        case "update":
+            return algosdk.OnApplicationComplete.UpdateApplicationOC;
+        case "optin":
+            return algosdk.OnApplicationComplete.OptInOC;
+        case "delete":
+            return algosdk.OnApplicationComplete.DeleteApplicationOC;
+        case "clear":
+            return algosdk.OnApplicationComplete.ClearStateOC;
+        case "closeout":
+            return algosdk.OnApplicationComplete.CloseOutOC;
+        default:
+            throw Error("did not recognize application operation string " + inString);
+    }
+}
+
+function splitAndProcessAppArgs(inArgs) {
+    let splitArgs = inArgs.split(",");
+    let subArgs = [];
+    splitArgs.forEach((subArg) => {
+       subArgs.push(subArg.split(":"));
+    });
+    let appArgs = [];
+    subArgs.forEach((subArg) => {
+       switch (subArg[0]) {
+           case "str":
+               appArgs.push(new Uint8Array(Buffer.from(subArg[1])));
+               break;
+           case "int":
+
+               appArgs.push(new Uint8Array([parseInt(subArg[1])]));
+               break;
+           case "addr":
+               appArgs.push(address.decode(subArg[1])["publicKey"]);
+               break;
+           default:
+               throw Error("did not recognize app arg of type" + subArg[0])
+       }
+    });
+    return appArgs;
+}
+
+When('I build an application transaction with operation {string}, application-id {int}, sender {string}, approval-program {string}, clear-program {string}, global-bytes {int}, global-ints {int}, local-bytes {int}, local-ints {int}, app-args {string}, foreign-apps {string}, app-accounts {string}, fee {int}, first-valid {int}, last-valid {int}, genesis-hash {string}', function (operationString, appIndex, sender, approvalProgramFile, clearProgramFile, numGlobalByteSlices, numGlobalInts, numLocalByteSlices, numLocalInts, appArgsCommaSeparatedString, foreignAppsCommaSeparatedString, appAccountsCommaSeparatedString, fee, firstValid, lastValid, genesisHashBase64) {
+    // operation string to enum
+    let operation = operationStringToEnum(operationString);
```

Comment:
Unfortunately, I think that each of these should use the new "makeApplicationXXX" functions. That way if you had accidentally set a wrong fields in one of those helpers the test would catch it.